pycc/cclambda.py

```diff
@@ -129,8 +129,14 @@ def solve_lambda(self, e_conv=1e-7, r_conv=1e-7, maxiter=100, max_diis=8, start_
             Wabef_cc3 = self.build_cc3_Wabef(o, v, ERI, t1)
 
             # Building intermediates in t3l1
-            Zmndi = np.zeros((no,no,nv,no))
-            Zmdfa = np.zeros((no,nv,nv,nv))
+            if isinstance(t1, torch.Tensor):                
+                Zmndi = torch.zeros_like(t2[:,:,:,:no])
+                Zmdfa = torch.zeros_like(t2)
+                Zmdfa = torch.nn.functional.pad(Zmdfa, ((0,0),(0,nv-no),(0,0),(0,0)))
+            else:
+                Zmndi = np.zeros_like(t2[:,:,:,:no])
+                Zmdfa = np.zeros_like(t2)
+                Zmdfa = np.pad(Zmdfa, ((0,0),(0,nv-no),(0,0),(0,0)))
```

Comment:
np.pad() (torch.nn.functional.pad()) is used to reshape the tensors when having not enough elements in the original shape.